src/execute/executeAndForwardOutput.js

```diff
@@ -23,11 +28,15 @@ async function executeAndForwardOutput(code, next) {
       commandOutput += data;
     });
 
+    // child.on('close', (code) => {
+    //   next(code, commandOutput);
+    // });
     child.on('close', (code) => {
-      next(code, commandOutput);
+      res.status(200).json({ code, output: commandOutput });
     });
   } catch (err) {
-    console.error(err);
+    // console.log(err);
+    res.status(500).json({ error: err.message });
```

Comment:
res is not available here. I guess this continuation-passing style is misleading. Maybe we will need a refactor.

To solve the problem, you have to edit the execute handler too.

 VITE v4.3.9  ready in 426 ms

  ➜  Local:   http://localhost:5173/
  ➜  Network: use --host to expose
file:///Users/ko/Documents/GitHub/Junior/src/execute/executeAndForwardOutput.js:39
    res.status(500).json({ error: err.message });
    ^
ReferenceError: res is not defined
    at executeAndForwardOutput (file:///Users/ko/Documents/GitHub/Junior/src/execute/executeAndForwardOutput.js:39:5)
    at executeHandler (file:///Users/ko/Documents/GitHub/Junior/src/backend/handlers/executeHandler.js:4:3)

Reply:
Oh, I forgot to change it. If logs would do for now uncommenting console. error(error) would do.

Reply:
<img width="924" alt="Screenshot 2023-07-30 005312" src="https://github.com/tisztamo/Junior/assets/76871537/afecf9e3-d768-487f-899e-9dce090a9068">

I have edited the handler, the function can use res as a parameter. Let me know if the edit is correct.

Reply:
The problem is that executeAndForwardOutput is also called from the command line interface where there is no res, so it needs to return the result. E.g. we have

```js
const executeCode = async (code) => {
  confirmExecution(code, () => executeAndForwardOutput(code, startInteractiveSession));
}
```

Instead of returning the result, executeAndForwardOutput calls the function it got in the "next" argument with the result. This is called continuation passing style and I think it was introduced by ChatGPT during a refactor of a strange recursive function that was coming from ChatSH (and also generated by ChatGPT I guess). I never liked it, and now I see we have to get rid of it. Just opened #29

I also have to say sorry as I mistakenly thought this one is a good first issue. It wasn't.

I still think that the version you screenshoted first in the issue may work well, although I haven't seen its code. I would like to ask you to push a version where the `next(code, commandOutput);` line is not modified, just the try-catch block expanded, and no change in the handler or the frontend. Can you make it?

Reply:
Yeah sure I will do it .

Reply:
I would like to dive deep in this issue,  I think I'll learn, will go through cli part and try to understand.